packages/flutter/lib/src/widgets/image.dart

```diff
@@ -436,6 +436,12 @@ class Image extends StatefulWidget {
   /// will be rendered to the constraints of the layout or [width] and [height]
   /// regardless of these parameters. These parameters are primarily intended
   /// to reduce the memory usage of [ImageCache].
+  ///
+  /// Loading an image from a file keeps a cache of the file.
+  /// So, if you save an image to a file and then change the underlying image
+  /// file with the same name, the original image remains because of caching.
+  /// Therefore, if you want to change the image with a different file,
+  /// you should manually evict the file from the cache.
```

Comment:
nit: I find the phrasing of this slightly confusing/misleding - e.g. where is the cache?

Perhaps something like this:

```
Loading an image from a file creates an in memory copy of the file, which is retained
in the [ImageCache]. The underlying file is not monitored for changes. If it does change, 
the application should evict the entry rom the [ImageCache].
```

We should add similar documentation on `FileImage` in the painting binding.

Reply:
We should also explain _how_ to evict the image - which is a little difficult to do with an [Image.file] constructed widget, and so we should probably explain that if you expect the underlying data to chagne you should directly work with a `FileImage` provider so it can more easily be evicted on changes.